odc_k8s/provider.tf

```diff
@@ -27,4 +27,110 @@ provider "helm" {
     token                  = data.aws_eks_cluster_auth.cluster.token
   }
   install_tiller = false
+}
+
+# null_resource.install_kubectl is a workaround for limitations in the official provider.kubernetes
+# This will install kubectl and awscli and initialise it for use with the deployed cluster
+# kubectl can then be used via local-exec commands to update k8s configuration for those cases where the
+# official provider.kubernetes is not sufficient (e.g. CRD configuration)
+# 
+# Implementation Notes:
+#   + It is essential any resource that use kubectl be added to the triggers block or kubectl will not be installed on any updates
+#   + resources using kubectl need to include provisioner blocks for plan and destroy so they clean up after themselves and apply updates
+#
+# Reference: https://github.com/cloudposse/terraform-aws-eks-cluster/blob/master/auth.tf
+variable "install_aws_cli" {
+  type        = bool
+  default     = false
+  description = "Set to `true` to install AWS CLI if the module is provisioned on workstations where AWS CLI is not installed by default, e.g. Terraform Cloud workers"
+}
+
+variable "install_kubectl" {
+  type        = bool
+  default     = false
+  description = "Set to `true` to install `kubectl` if the module is provisioned on workstations where `kubectl` is not installed by default, e.g. Terraform Cloud workers"
+}
+
+variable "kubectl_version" {
+  type        = string
+  default     = ""
+  description = "`kubectl` version to install. If not specified, the latest version will be used"
+}
+
+variable "external_packages_install_path" {
+  type        = string
+  default     = ""
+  description = "Path to install external packages, e.g. AWS CLI and `kubectl`. Used when the module is provisioned on workstations where the external packages are not installed by default, e.g. Terraform Cloud workers"
+}
+
+variable "aws_eks_update_kubeconfig_additional_arguments" {
+  type        = string
+  default     = ""
+  description = "Additional arguments for `aws eks update-kubeconfig` command, e.g. `--role-arn xxxxxxxxx`. For more info, see https://docs.aws.amazon.com/cli/latest/reference/eks/update-kubeconfig.html"
+}
+
+variable "local_exec_interpreter" {
+  type        = string
+  default     = "/bin/bash"
+  description = "shell to use for local exec"
+}
+
+variable "kubeconfig_path" {
+  type        = string
+  default     = "~/.kube/config"
+  description = "The path to `kubeconfig` file"
+}
+
+locals {
+  external_packages_install_path = var.external_packages_install_path == "" ? join("/", [abspath(path.module), ".terraform/bin"]) : var.external_packages_install_path
+  kubectl_version                = var.kubectl_version == "" ? "$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)" : var.kubectl_version
+
+  cluster_name = data.aws_eks_cluster.cluster.id
+ }
+
+resource "null_resource" "install_kubectl" {
```

Comment:
this resource installs both aws-cli and kubectl:
can we name it better? 
or 
create two separate resources - "install_aws_cli" and "install_kubectl" ?

Reply:
aws_cli is only installed because it is required for kubectl installation. so the objective of the script is to install kubectl.
I've also refactored this and it is now purely an installation script called install_kubectl as part of install_extra_software. I think it will be valid to keep this altogether rather than split it out so it does everything that is required and not just part of it.
The only reason any of it can be turned on/off for install is because I don't want it to break what the old "you must have this installed locally" behavior